test/VisualRecognition.v4.UnitTests/VisualRecognitionUnitTests.cs

```diff
@@ -0,0 +1,1491 @@
+                Result = new AnalyzeResponse()
+                {
+                    Images = new List<Image>()
+                    {
+                        new Image()
+                        {
+                            Source = new ImageSource()
+                            {
+                                Type = "type",
+                                Filename = "filename",
+                                ArchiveFilename = "archiveFilename",
+                                SourceUrl = "sourceUrl",
+                                ResolvedUrl = "resolvedUrl"
+                            },
+                            Dimensions = new ImageDimensions()
+                            {
+                                Width = 100,
+                                Height = 200
+                            },
+                            Objects = new DetectedObjects()
+                            {
+                                Collections = new List<CollectionObjects>()
+                                {
+                                    new CollectionObjects()
+                                    {
+                                        CollectionId = "collectionId",
+                                        Objects = new List<ObjectDetail>()
+                                        {
+                                            new ObjectDetail()
+                                            {
+                                                _Object = "object",
+                                                Location = new Location()
+                                                {
+                                                    Top = 0,
+                                                    Left = 1,
+                                                    Width = 2,
+                                                    Height = 3
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+            #endregion
+
+            NoAuthAuthenticator authenticator = new NoAuthAuthenticator();
+            VisualRecognitionService service = new VisualRecognitionService("versionDate", authenticator);
+            service.Analyze(
+                collectionIds: "colletionIds",
+                features: "features");
+
+            //Assert.
+        }
+        #endregion
+
+        #region Create Client
+        private IClient CreateClient()
+        {
+            IClient client = Substitute.For<IClient>();
+
+            client.WithAuthentication(Arg.Any<string>(), Arg.Any<string>())
+                .Returns(client);
+
+            return client;
+        }
+        #endregion
+
+        #region Environments
+        //#region List Environments

```

Comment:
Looks like everything under this has to do with Discovery haha

Reply:
ugh... my unit tests got lost somewhere :(

Reply:
Nooo 😭😭

Reply:
:( seriously... thats what i get for working on multiple machines. Ill try to find them. If not, ill include them in unit test generation!